modules/apps/portal-template/portal-template-react-renderer-api/src/main/java/com/liferay/portal/template/react/renderer/ComponentDescriptor.java

```diff
@@ -41,6 +42,16 @@ public ComponentDescriptor(
 		String module, String componentId, Collection<String> dependencies,
 		boolean positionInLine) {
 
+		this(module, componentId, dependencies, positionInLine, null, null);
+	}
+
+	public ComponentDescriptor(
+		String module, String componentId, Collection<String> dependencies,
+		boolean positionInLine, String additionalProps,
+		Map<String, String> additionalPropsData) {
```

Comment:
As I did originally... we could just abuse the existing `dependencies` argument and give it a different meaning in this context. After all, we're not using it for anything and I'm not sure how deps could be passed in a way that make sense for this.

In any case, I decided to err on the caution side and simply add additional params in case we ever get to use `dependencies` as intended.

Reply:
Yeah, I have my doubts if that dependencies argument was ever really needed since the module entry point can import whatever it needs. Maybe it was for the Soy components.

Reply:
One thing that concerns me is that by reading `additionalProps` it's not clear that this is actually a path to module. Maybe `additionalPropsModule` or following @wincent's suggestion of `propsTransformer` that could be `propsTransformerModule` in the java domain since we already have `String module` here.

Reply:
![transformer](https://user-images.githubusercontent.com/905006/84911326-4a957000-b0b8-11ea-9d04-2098331337a9.gif)